atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -788,12 +788,18 @@ private void extractLocalWritesForRow(
     @Override
     @Idempotent
     public ListenableFuture<Map<Cell, byte[]>> getAsync(TableReference tableRef, Set<Cell> cells) {
-        return cache.get()
-                .getAsync(
-                        tableRef,
-                        cells,
-                        (table, uncached) -> getInternal(
-                                "getAsync", tableRef, uncached, keyValueService, defaultTransactionService));
+        return Futures.transform(
+                cache.get()
+                        .getAsync(
+                                tableRef,
+                                cells,
+                                (table, uncached) -> getInternal(
+                                        "getAsync", tableRef, uncached, keyValueService, defaultTransactionService)),
+                input -> {
+                    ensureStillRunning();
+                    return input;
+                },
```

Comment:
technically, this is the output / result, not the input

Reply:
Also, let's extract a `scopeToTransaction` method for futures too, then it is consistent and a bit cleaner.